### ISSUE TITLE
respect __init_subclass__ in _ModuleMeta

### ISSUE THREAD

**Issue by Roger-luo** [patrick-kidger/equinox#520]
Currently `_ModuleMeta` does not forward the arguments of `__init_subclass__` so as a result it's not working if you write something like

```python
class Bar(eqx.Module):
    
    def __init_subclass__(cls, a, b):
        def get_a(self) -> int:
            return 1
        setattr(cls, "get_a", property(get_a))

class Foo(Bar, a=1, b=2):
    pass
```

**Comment by Roger-luo** [comment]
For context, I'm using this to auto-generate some methods for subclass. And this works with `flax.struct.dataclass` but not `eqx.Module`, without supporting I have to do these work manually for all subclass which is kinda manual

**Comment by patrick-kidger** [comment]
I think this should already be fixed on the `dev` branch. (See also #481.) I'm planning on doing a release this week.

**Comment by patrick-kidger** [comment]
Now resolved in the v0.11.0 release!
